From f8d3a7e68a0245a05683020a81cdbe55f29dc533 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:00:27 -0400
Subject: [acpi] dock: Drop unnecessary local variable from dock_add()

Message-id: <20140710010026.22146.808.stgit@gir.stowe>
Patchwork-id: 85560
O-Subject: [RHEL7.1 PATCH 021/105] ACPI / dock: Drop unnecessary local variable from dock_add()
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 2efbca4dfc7b43951de6dd1647f9eebda9d4372b

commit 2efbca4dfc7b43951de6dd1647f9eebda9d4372b
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Jul 5 03:23:36 2013 +0200

    ACPI / dock: Drop unnecessary local variable from dock_add()

    The local variable id in dock_add() is not necessary, so drop it.

    While we're at it, use an initializer to clear the local variable ds
    and drop the memset() used for this purpose.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index c89a9c3..f601658 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -836,14 +836,13 @@ static struct attribute_group dock_attribute_group = {
  */
 static int __init dock_add(acpi_handle handle)
 {
- int ret, id;
- struct dock_station ds, *dock_station;
+ struct dock_station *dock_station, ds = { NULL, };
  struct platform_device *dd;
  acpi_status status;
+ int ret;
 
- id = dock_station_count;
- memset(&ds, 0, sizeof(ds));
- dd = platform_device_register_data(NULL, "dock", id, &ds, sizeof(ds));
+ dd = platform_device_register_data(NULL, "dock", dock_station_count,
+        &ds, sizeof(ds));
  if (IS_ERR(dd))
   return PTR_ERR(dd);
 
-- 
1.7.1